src/controller-wrapper.js

```diff
@@ -58,18 +75,18 @@ define([
          *
          * @return {Object} original or modified clone of controller info
          */
-        prepareControllerInfo : function() {
+        prepareControllerInfo : function(controllerInfo) {
```

Comment:
What worries me is that the  `controllerInfo` object is modified here, as we share the same reference as the `PageManager`, but this class does not own that object.

In other words, we're giving this class the responsibility to modify the routing behavior, which should only be a PageManager's concern, and if someone else takes this as an example, it might lead to a routing flow that's harder to maintain, as it doesn't own its state.

Could you please change this function to return an object like `{ rootId : this.currentController }` and perform the `extend` in the PageManager? so we can avoid side effects here :)